### ISSUE TITLE
window.on('destroy', handler)

### ISSUE THREAD

**Issue by samuelmr** [pebble/pebblejs#19]
add a event that fires when a window is destroyed. Use case: I start navigator.geolocation.watchPosition() when I initiate a window, and use it to populate a field in the window. When the user leaves the window (opens a new window or goes back to previous), there’s no more need to watch position. How should the developer know when the window is no longer active?

Could you add e.g. window.on('destroy', handler) or window.ondestroy(handler) functionality to all window objects (also to cards and menus)?

**Comment by Meiguro** [comment]
Nice catch! For some reason the WindowStack receives the show and hide events but don't pass it further. I'll be sure to change that.

**Comment by samuelmr** [comment]
I suppose this works now. I'm closing the issue.

**Comment by rajington** [comment]
I'm not sure if it is working, at least not on CloudPebble. I tried to write the simplest application, a select triggers a window show that has a listener on it, unfortunately it never gets triggered. My code below:

```javascript
var UI = require('ui');

var main = new UI.Card({
  title: 'click select and view the logs'
});

var wind = new UI.Window();
wind.on('show', function() {
  console.log('Window is shown!');
});

main.show();

main.on('click', 'select', function(e) {
  console.log('Window is selected!');
  wind.show();
});
```

results in the following code when I repeatedly show and hide the window.

```
[PHONE] pebble-app.js:?: Window is selected!
[PHONE] pebble-app.js:?: (+) [window 2] : [card 1],[window 2]
[PHONE] pebble-app.js:?: (-) [window 2] : [card 1]
[PHONE] pebble-app.js:?: Window is selected!
[PHONE] pebble-app.js:?: (+) [window 2] : [card 1],[window 2]
[PHONE] pebble-app.js:?: (-) [window 2] : [card 1]
[PHONE] pebble-app.js:?: Window is selected!
[PHONE] pebble-app.js:?: (+) [window 2] : [card 1],[window 2]
```

**Comment by Meiguro** [comment]
Right, it is not yet on CloudPebble. I will notify this ticket when it is.
